lib/fluent/plugin/out_datadog.rb

```diff
@@ -300,18 +301,28 @@ class DatadogHTTPClient < DatadogClient
     require 'net/http'
     require 'net/http/persistent'
 
-    def initialize(logger, use_ssl, no_ssl_validation, host, ssl_port, port, use_compression, api_key)
+    def initialize(logger, use_ssl, no_ssl_validation, host, ssl_port, port, proxy, use_compression, api_key)
       @logger = logger
       protocol = use_ssl ? "https" : "http"
       port = use_ssl ? ssl_port : port
       @uri = URI("#{protocol}://#{host}:#{port.to_s}/v1/input/#{api_key}")
+      proxy_uri = :ENV
+      if proxy
+        proxy_uri = URI.parse(proxy)
+      elsif ENV['HTTP_PROXY'] || ENV['http_proxy']
+        logger.info("Using HTTP proxy defined in `HTTP_PROXY`/`http_proxy` env vars")
+      end
       logger.info("Starting HTTP connection to #{protocol}://#{host}:#{port.to_s} with compression " + (use_compression ? "enabled" : "disabled"))
-      @client = Net::HTTP::Persistent.new name: "fluent-plugin-datadog-logcollector"
+      @client = Net::HTTP::Persistent.new name: "fluent-plugin-datadog-logcollector", proxy: proxy_uri
```

Comment:
When `proxy` is nil, is the `Net::HTTP::Persistent.new` implementation automatically looking for `proxy_uri["HTTP_PROXY"]`? With `proxy` nil, I don't get where's the magic with L309 & here where we assume that it will use the env vars if set.

Reply:
Nevermind, you documented it in the PR description.

Reply:
(this is also covered in a unit test FWIW)